src/components/switch/switch.component.ts

```diff
@@ -179,46 +186,69 @@ export default class SlSwitch extends ShoelaceElement implements ShoelaceFormCon
   }
 
   render() {
+    const hasHelpTextSlot = this.hasSlotController.test('help-text');
+    const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
+
     return html`
-      <label
-        part="base"
+      <div
         class=${classMap({
-          switch: true,
-          'switch--checked': this.checked,
-          'switch--disabled': this.disabled,
-          'switch--focused': this.hasFocus,
-          'switch--small': this.size === 'small',
-          'switch--medium': this.size === 'medium',
-          'switch--large': this.size === 'large'
+          'form-control': true,
+          'form-control--small': this.size === 'small',
+          'form-control--medium': this.size === 'medium',
+          'form-control--large': this.size === 'large',
+          'form-control--has-help-text': hasHelpText
         })}
       >
-        <input
-          class="switch__input"
-          type="checkbox"
-          title=${this.title /* An empty title prevents browser validation tooltips from appearing on hover */}
-          name=${this.name}
-          value=${ifDefined(this.value)}
-          .checked=${live(this.checked)}
-          .disabled=${this.disabled}
-          .required=${this.required}
-          role="switch"
-          aria-checked=${this.checked ? 'true' : 'false'}
-          @click=${this.handleClick}
-          @input=${this.handleInput}
-          @invalid=${this.handleInvalid}
-          @blur=${this.handleBlur}
-          @focus=${this.handleFocus}
-          @keydown=${this.handleKeyDown}
-        />
-
-        <span part="control" class="switch__control">
-          <span part="thumb" class="switch__thumb"></span>
-        </span>
-
-        <div part="label" class="switch__label">
-          <slot></slot>
+        <label
+          part="base"
```

Comment:
Moving `part="base"` to the containing `<div>` instead of  keeping it on `<label>` would be ideal and in line with the other components.

However, it would also be a breaking change. A consumer, for example, may be applying `display: grid` to the base part and resizing or rearranging the grid assuming certain grid items and in a certain order.

Any thoughts?

Reply:
I agree it's a breaking to move the `part="base"` to the encapsulating `<div>`, however, by not moving it, we're breaking a lot of existing conventions in the library which I don't particularly love.

I'm gonna talk with @claviska , but this may be a good candidate for a revamp in the upcoming 3.0.

Reply:
Yeah this change is probably better for 3.x.